### PR TITLE
ENH: Added clearer exception for np.diff on 0-dimensional ndarray

### PR DESCRIPTION
**Pull request by navneet-nmk**
Referencing #13296 

Added TypeError message in np.diff for handling 0-dimensional ndarrays.

### CODE REVIEW

**Comment by mattip** [comment]
Please add a test that the exception type matches expectations to the `TestDiff` class in `numpy/lib/test/test_function_base.py`

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -1235,6 +1235,8 @@ def diff(a, n=1, axis=-1, prepend=np._NoValue, append=np._NoValue):
 
     a = asanyarray(a)
     nd = a.ndim
+    if nd == 0:
+        raise TypeError("diff requires more than one element")
```
Comment (eric-wieser) [inline numpy/lib/function_base.py]: This message is very wrong
```python
arr_0d = np.array(1)
arr_1d_empty = np.array([])

assert arr_1d_empty.size == 0  # has 0 elements
np.diff(arr_1d_empty) # works fine

assert arr_0d.size > 0  # has more than 0 elements
np.diff(arr_0d) # fails
```

**Review by eric-wieser — CHANGES_REQUESTED** [review]
This new error message is not correct. How about just `"Input must be at least 1D"`?

**Comment by navneet-nmk** [comment]
Pushed the changes as requested

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -1235,6 +1235,8 @@ def diff(a, n=1, axis=-1, prepend=np._NoValue, append=np._NoValue):
 
     a = asanyarray(a)
     nd = a.ndim
+    if nd == 0:
+        raise TypeError("diff requires input that is at least one dimensional")
```
Comment (eric-wieser) [inline numpy/lib/function_base.py]: I think this should be a `ValueError`?

**Comment by mattip** [comment]
I changed the error type and squashed the commits

**Comment by eric-wieser** [comment]
Thanks @navneet-nmk and Matti

**Pull request merged**